vizgrimoire/analysis/events_punchcard.py

```diff
@@ -102,18 +105,18 @@ def result(self):
             current_group = group
             if current_group <> old_group:
                 # we need to restart the process
-                punchcard[old_group] = week_structure.copy()
+                punchcard[old_group] = deepcopy(week_structure)
                 # Init data structure
-                week_structure = empty_week_structure.copy()
+                week_structure = deepcopy(empty_week_structure)
 
             weekday = int(data["weekday"][count])
             hour = int(data["hour"][count])
             mean = float(data["mean"][count])
             week_structure[weekday][hour] = mean
             count = count + 1
-            if count == len(data["group_name"]) -1:
```

Comment:
You need to make a copy of the last group at some point. Either you do it in that loop or out of the main for. But you need that copy ;).

In addition, having a list of 1 element would force that copy indeed. 'count' would be 0 and 'len' of a list of 1 element would be 1.